src/types/JSONSchema.ts

```diff
@@ -93,10 +97,18 @@ export interface EnumJSONSchema extends NormalizedJSONSchema {
   enum: any[]
 }
 
-export interface NamedEnumJSONSchema extends NormalizedJSONSchema {
+interface DeprecatedNamedEnumJSONSchema extends NormalizedJSONSchema {
+  'x-enum-varnames'?: never
   tsEnumNames: string[]
```

Comment:
Normalized schemas shouldn't contain `tsEnumNames`, since you're normalizing it away.

Reply:
So just deleted `DeprecatedNamedEnumJSONSchema` and rename `_NamedEnumJSONSchema`  to `NamedEnumJSONSchema`?

Reply:
Yeah just add `x-enum-varnames: string[]` and `tsEnumNames: never` to `NormalizedJSONSchema`. No need for new types if you can avoid it.